crates/astria-core/src/protocol/transaction/v1alpha1/action.rs

```diff
@@ -694,26 +694,26 @@ enum MintActionErrorKind {
 ///
 /// It also contains a `return_address` field which may or may not be the same as the signer
 /// of the packet. The funds will be returned to the `return_address` in the case of a timeout.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
All fields should be private.

Reply:
all the other actions have pub fields - we should probably make constructors for them instead? but a bit out of scope for here

Reply:
can we make a follow up issue here? seems a general code quality thing

Reply:
Gonna let it pass here because the other actions also have their fields exposed. But yeah, in general I prefer to not leak that stuff. We should address this in a follow, prior to publishing this crate.